src/app/components/client/InputField.tsx

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import { ReactNode, useRef } from "react";
+import { AriaTextFieldProps, useTextField } from "react-aria";
+import styles from "./InputField.module.scss";
+
+export type Props = {
+  children: ReactNode;
+};
+
+export const InputField = (props: AriaTextFieldProps) => {
+  const { errorMessage, isRequired, label, validationState, value } = props;
+  const inputRef = useRef(null);
+  const { errorMessageProps, inputProps, labelProps } = useTextField(
+    props,
+    inputRef
+  );
+  const isInvalid = validationState === "invalid";
+  const showError = errorMessage && isInvalid;
+
+  return (
+    <div className={styles.input}>
+      <label {...labelProps} className={styles.inputLabel}>
+        {label}
+        {isRequired ? "*" : ""}
```

Comment:
I wonder if there's a way we can indicate what this means... Perhaps something like this?

```suggestion
        {isRequired ? "<abbr title={l10n.getString("input-required")}>*</abbr>" : ""}
```

There are also some hits [here](https://duckduckgo.com/?t=ffab&q=accessible+required+asterisk&atb=v380-1&ia=web), including one by Dequeue, who make axe, so that might be worth checking out.

Reply:
Using `<abbr>` seemed like a good idea, but upon testing it this has the disadvantage that a screen reader will announce our custom message as well. React Aria seems to already add a `required` tag that is being announced. Thanks for linking the article — I checked the approaches listed there out. I think just hiding the asterisk from the screen reader is working well for now unless we would like to customize the “required message”: [710074e](https://github.com/mozilla/blurts-server/pull/3208/commits/710074ee14934c7e86540b005396b22f0d7b7618).